### ISSUE TITLE
unnecessary_debug_formatting does not respect MSRV

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#14952]
### Summary

I have a crate with MSRV 1.76.0 which is triggering unnecessary_debug_formatting when using debug formatting on a Path and OsStr. This lint should not fire since the display methods were not stabilized until Rust 1.87.0.

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text

```

**Comment by lopopolo** [comment]
As an aside, this lint is firing within a Manual implementation of fmt::Debug on a struct, which feels like a false positive.

**Comment by profetia** [comment]
Reproducer: 
```rust
#![warn(unnecessary_debug_formatting)]

use std::path::Path;

struct Foo(Path);

impl std::fmt::Debug for Foo {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{:?}", &self.0)
    }
}
```
